scripts/build-bootstraps.sh

```diff
@@ -385,27 +526,71 @@ main() {
-			echo "Supported architectures: '${TERMUX_DEFAULT_ARCHITECTURES[*]}'" 1>&2
+	if [[ "$TERMUX_PACKAGE_MANAGER" == *" "* ]] || [[ " ${TERMUX_SUPPORTED_PACKAGE_MANAGERS[*]} " != *" $TERMUX_PACKAGE_MANAGER "* ]]; then
+		echo "[!] Invalid package manager '$TERMUX_PACKAGE_MANAGER'" 1>&2
+		echo "Supported package managers: '${TERMUX_SUPPORTED_PACKAGE_MANAGERS[*]}'" 1>&2
+		exit 1
+	fi
+
+	for package_arch in "${TERMUX_ARCHITECTURES[@]}"; do
+		if [[ "$package_arch" == *" "* ]] || [[ " ${!TERMUX_SUPPORTED_ARCHITECTURES_TO_ABIS_MAP[*]} " != *" $package_arch "* ]]; then
+			echo "[!] Unsupported architecture '$package_arch' in architectures list: '${TERMUX_ARCHITECTURES[*]}'" 1>&2
+			echo "Supported architectures: '${!TERMUX_SUPPORTED_ARCHITECTURES_TO_ABIS_MAP[*]}'" 1>&2
 			return 1
 		fi
 	done
 
-	for TERMUX_ARCH in "${TERMUX_ARCHITECTURES[@]}"; do
-		termux_step_handle_buildarch
+	echo "[*] Building bootstraps for archs: ${TERMUX_ARCHITECTURES[*]}"
+	boostraps_build_start_time="$(date +%s)"
+
+	if [[ $NO_CLEAN_BEFORE_BUILD != "true" ]]; then
+		# The files under TERMUX_BUILT_PACKAGE_FILES_DIRECTORY
+		# should be deleted before building for an arch
+		# since if the output directory already has older
+		# (deb) package files with multiple versions of the same
+		# package, an error would be raised. And even if there
+		# was a single deb for the package, it may be the wrong
+		# version.
+		if [ -d "$TERMUX_BUILT_PACKAGE_FILES_DIRECTORY" ]; then
+			find "$TERMUX_BUILT_PACKAGE_FILES_DIRECTORY" -maxdepth 1 -type f -delete
+		fi
+
+		# Delete any old bootstraps for all architectures.
+		if [ -d "$TERMUX_BOOTSTRAP_OUTPUT_DIRECTORY" ]; then
+			find "$TERMUX_BOOTSTRAP_OUTPUT_DIRECTORY" -maxdepth 1 -type f \
+				\( -name "bootstrap-aarch64.zip" -o -name "bootstrap-arm.zip" -o -name "bootstrap-x86_64.zip" -o -name "bootstrap-i686.zip" \) \
+				-delete
+		fi
+	fi
+
+	mkdir -p "$TERMUX_PACKAGES_REPO_OUTPUT_DIRECTORY"
+
+	for package_arch in "${TERMUX_ARCHITECTURES[@]}"; do
+		echo $'\n\n\n\n\n'"[*] Starting to build bootstrap for arch '$package_arch'..."
+		boostrap_build_start_time="$(date +%s)"
+
+		TERMUX_ARCH="$package_arch" termux_step_handle_buildarch
 
-		if [[ $FORCE_BUILD_PACKAGES == "1" ]]; then
-			rm -f "$TERMUX_BUILT_PACKAGES_DIRECTORY_FOR_ARCH"/*
-			rm -f "$TERMUX_BUILT_DEBS_DIRECTORY"/*
+		if [[ $NO_CLEAN_BEFORE_BUILD != "true" ]]; then
+			# The "-f" flag must never be passed to `build-package.sh`
+			# with `BUILD_PACKAGE_OPTIONS` as otherwise if a package
+			# in `PACKAGES` list first gets built as a dependency of
+			# another package, and then its build is manually
+			# started as part of bootstrap packages, then it will
+			# get built again, which may also fail or create an
+			# inconsistent state for certain packages as prefix
+			# wouldn't have been wiped. So all packages are manually
+			# cleaned before building bootstrap.
+
+			# Remove `$HOME/.termux-build` and termux prefix.
+			"$TERMUX_SCRIPTDIR/clean.sh"
 		fi
```

Comment:
Probably can be 
```suggestion
		# The "-f" flag must never be passed to `build-package.sh`
		# with `BUILD_PACKAGE_OPTIONS` as otherwise if a package
		# in `PACKAGES` list first gets built as a dependency of
		# another package, and then its build is manually
		# started as part of bootstrap packages, then it will
		# get built again, which may also fail or create an
		# inconsistent state for certain packages as prefix
		# wouldn't have been wiped. So all packages are manually
		# cleaned before building bootstrap.

		# Remove `$HOME/.termux-build` and termux prefix.
		[[ $NO_CLEAN_BEFORE_BUILD != "true" ]] && "$TERMUX_SCRIPTDIR/clean.sh"
```

Reply:
If some other code would need to be added with no clean enabled, then would have to readd the if condition, make changes to indent, which would also mess with `git` line history. Additionally, by newer build-bootstraps.sh script does not use `set -e` and can't use above directly with ` || return $?`.